### ISSUE TITLE
Add Boost download and compilation before use it

### ISSUE THREAD

**Issue by vicente-gonzalez-ruiz** [P2PSP/I-want-to-use-it#4]
There is a problem using Boost from the package repositories in some OS (for example, Ubuntu 16.04.2 LTS). We think that a good solution could be to include in the file `make.py` an option to compile `p2psp-console` using downloaded (and compiled) version of Boost. The compilation of Boost could be stored in a temporal directory inside `p2psp-console`.

**Comment by anu1097** [comment]
The compiled binary version of boost can be downloaded from here https://sourceforge.net/projects/boost/files/boost-binaries/. But after putting it in a separate directory in p2psp-console how do we link it in make.py ?

**Comment by vicente-gonzalez-ruiz** [comment]
Hi @anu1097 . Thanks for working in this issue. Two comments: (1) I think that it is preferable that we download the source and compile it, and (2) if I'm not wrong, it is possible to indicate to `cmake` where is Boost through the `CMakeLists.txt` commands:

```
set(BOOST_ROOT "/usr/local/boost_1_59_0")
set(Boost_NO_SYSTEM_PATHS TRUE)
set(BOOST_INCLUDEDIR "/usr/local/boost_1_59_0/include")
set(BOOST_LIBRARYDIR "/usr/local/boost_1_59_0/stage/lib")
```
where `/usr/local/boost_1_59_0` is the path to the downloaded (and compiled) Boost (don't use this version, use the last one).

But, notice that we don't want to add this stuff to this file. We prefer to manage all this only using `make.py`.

Thanks!

**Comment by abhay44** [comment]
@vicente-gonzalez-ruiz You want to download and compile the whole boost libraries using `make.py`? am I correct?

**Comment by vicente-gonzalez-ruiz** [comment]
@abhay44, right. If we download a compiled Boost we run the risk of getting the same errors :-/

**Comment by abhay44** [comment]
@vicente-gonzalez-ruiz there are lot of boost libraries and the compilation takes more time if we manually download and compile them.Is it ok? or do we have other option?

**Comment by vicente-gonzalez-ruiz** [comment]
Hi @abhay44 . Check https://github.com/P2PSP/core/pull/127

**Comment by vicente-gonzalez-ruiz** [comment]
@abhay44 , check https://github.com/P2PSP/core/pull/127

**Comment by abhay44** [comment]
@vicente-gonzalez-ruiz I think we can close this issue. Fixed in [P2PSP/core#127](https://github.com/P2PSP/core/pull/127).

**Comment by vicente-gonzalez-ruiz** [comment]
Right. Please, do that.
